File: ProvincialEnterprise/Custom/Branch/branchController.js
app.controller("branchCtrl", function ($scope, branchService) {

    getAllCompanies();

    $scope.getAllBranchesByCompany = function () {
        var companyId = $scope.companyId;
        var getSectionData = branchService.getAllBranchesByCompany(companyId);
        getSectionData.then(function (branch) {
            $scope.branches = branch.data;
        }, function () {
            alert('Error in getting records');
        });
    }

    function getAllCompanies() {
        var getBranchData = branchService.getCompanies();
        getBranchData.then(function (company) {
            $scope.companies = company.data;
        }, function () {
            alert("Error getting record");
        });
    }

    $scope.editBranch = function (branch) {
        var getBranchData = branchService.getBranch(branch.BranchId);
        getBranchData.then(function (_branch) {

            $scope.branch = _branch.data;
            $scope.branchId = branch.BranchId;
            $scope.name = branch.Name;
            $scope.address = branch.Address;
            $scope.contact = branch.Contact;
            $scope.web = branch.Web;
            $scope.email = branch.Email;
            $scope.companyId = branch.CompanyId;
            $scope.Action = "Update";
        }, function () {
            alert('Error in getting records');
        });
    }

    $scope.addUpdateBranch = function () {
        var Branch = {
            Name: $scope.name,
            Address: $scope.address,
            Contact: $scope.contact,
            Web: $scope.web,
            Email: $scope.email,
            CompanyId: $scope.companyId
        };
        var getBranchAction = $scope.Action;
        if (getBranchAction == "Update") {
            Branch.BranchId = $scope.branchId;
            var getBranchData = branchService.updateBranch(Branch);
            getBranchData.then(function(msg) {
                $scope.getAllBranchesByCompany();
                alert(msg.data);
            }, function() {
                alert('Error in updating record');
            });
        } else {
            var getBranchData = branchService.addBranch(Branch);
            getBranchData.then(function(msg) {
                $scope.getAllBranchesByCompany();
                alert(msg.data);
                ClearFields();
            },function() {
                alert('Error in adding record');
            });
        }
    }

    $scope.deleteBranch = function(branch) {
        var getBranchData = branchService.deleteBranch(branch.BranchId);
        getBranchData.then(function(msg) {
            $scope.getAllBranchesByCompany();
            alert(msg.data);
            ClearFields();
        }, function() {
            alert('Error in deleting record');
        });
    }

    $scope.AddBranchDiv = function () {
        ClearFields();
    }

    function ClearFields() {
        $scope.branchId = "";
        $scope.name = "";
        $scope.address = "";
        $scope.contact = "";
        $scope.web = "";
        $scope.email = "";
        $scope.Action = "Add";
       // $scope.companyId = "";
        //getAllCompanies();
    }

    $scope.Cancel = function () {
        ClearFields();
    };
});

File: ProvincialEnterprise/Controllers/BranchController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using ProvincialEnterprise.Models;
using ProvincialEnterprise.Models.Repository;

namespace ProvincialEnterprise.Controllers
{
    public class BranchController : Controller
    {
        BranchRepository repository = new BranchRepository();
        CompanyRepository compRepository = new CompanyRepository();
        // GET: Branch
        public ActionResult Index()
        {
            return View();
        }

        public JsonResult GetAllBranches()
        {
            var branches = repository.GetBranches();
            return Json(branches, JsonRequestBehavior.AllowGet);
        }

        public JsonResult GetAllBranchesByCompany(string id)
        {
            if (id != "null")
            {
                int companyId = Convert.ToInt32(id);
                var branches = repository.GetBranchesByCompany(companyId);
                return Json(branches, JsonRequestBehavior.AllowGet);
            }
            else
                return null;
        }

        public JsonResult GetAllCompanies()
        {
            var companies = compRepository.GetCompanies();
            return Json(companies, JsonRequestBehavior.AllowGet);
        }


        public JsonResult GetBranchById(string id)
        {
            if (id != null)
            {
                int branchId = Convert.ToInt32(id);
                var branch = repository.GetBrancheById(branchId);
                return Json(branch, JsonRequestBehavior.AllowGet);
            }
            else
            {
                return null;
            }
        }

        //public string GetDuplicate(string companyId, string name)
        //{
        //    if (repository.GetDuplicate(companyId,name))
        //    {
        //        return "Duplicate Data.";
        //    }
        //    else
        //        return null;
        //}

        public string UpdateBranch(Branch branch)
        {
            if (branch != null)
            {
                int branchId = Convert.ToInt32(branch.BranchId);
                Branch branchObj = repository.GetById(branchId);
                branchObj.Name = branch.Name;
                branchObj.Address = branch.Address;
                branchObj.Contact = branch.Contact;
                branchObj.Web = branch.Web;
                branchObj.Email = branch.Email;
                branchObj.CompanyId = branch.CompanyId;
                repository.SaveChanges();
                return "Record updated successfully";
            }
            else
            {
                return "Invalid record";
            }
        }

        public string AddBranch(Branch branch)
        {
            if (branch != null)
            {
                if (repository.GetDuplicate(Convert.ToInt32(branch.CompanyId), branch.Name))
                {
                    return "Duplicate record is found.";
                }
                else
                {
                    repository.Add(branch);
                    repository.SaveChanges();
                    return "Record added successfully";
                }
            }
            else
            {
                return "Invalid record";
            }
        }

        public string DeleteBranch(string id)
        {
            if (!String.IsNullOrEmpty(id))
            {
                try
                {
                    int branchId = Int32.Parse(id);

                    var branch = repository.Get(branchId);
                    repository.Delete(branch);
                    repository.SaveChanges();
                    return "Selected record deleted sucessfully";

                }
                catch (Exception)
                {
                    return "Branch details not found";
                }
            }
            else
            {
                return "Invalid operation";
            }
        }
    }
}